### DISCUSSION TITLE
Rasengan 1.0.0 Beta

### DISCUSSION THREAD

**Discussion by dilane3** [rasengan-dev/rasenganjs#3] [Announcements]
# Rasengan.js 1.0.0 Beta 🎉 

We are very excited to announce you the release of a new React Framework called **`Rasengan.js`**.

`Rasengan.js 1.0.0-beta` is the first beta version of the Rasengan.js framework. 

Based on modern tools, Rasengan.js will help you to create **high-quality web applications** with the benefit of React Components. It's a simple and easy-to-use framework that will help you to create a **beautiful and responsive web application**.

## Main features

Rasengan.js supports the following features:

- [**`Routing`**](https://rasengan.dev/docs/core/routing/base-concepts): Rasengan.js provides a simple and easy-to-use routing system that allows you to create routes for your web application.
- [**`CSS Support`**](https://rasengan.dev/docs/core/styling/css-module): Adding styles to your components is easy with Rasengan.js. You can use CSS Modules, tailwindCss, Preprocessors like Sass, Less and Stylus.
- [**`SSR`**](https://rasengan.dev/docs/core/rendering/introduction#server-side-rendering-ssr): Rasengan.js supports server-side rendering (SSR) out of the box. This rendering approach is useful for SEO and performance optimization.
- [**`Data Fetching`**](https://rasengan.dev/docs/core/rendering/data-fetching): Rasengan.js provides a simple and easy-to-use data fetching system that allows you to fetch data from an API.

Based on [Vite](https://vitejs.dev), it benefit of its high performance in terms of bundling, code splitting which makes it a great choice for your web application.

## Built with modern tools

Rasengan.js is built with modern tools like:

- [**`React`**](https://react.dev): A JavaScript library for building user interfaces.
- [**`Vite`**](https://vitejs.dev): A build tool that aims to provide a faster and leaner development experience for modern web projects.
- [**`React Router`**](https://reactrouter.com): A routing library for React.

## Why Rasengan.js?

Rasengan.js is a simple and easy-to-use framework that will help you to create a beautiful and responsive web application. It is built with modern tools like React, Vite, and React Router, which makes it a great choice for your web application.

The reason that the creator of Rasengan.js ([Dilane Kombou](https://twitter.com/dilanekombou)) decided to create this framework is because of the lack of performance while using only React and Vite. If you want to improve the performance of your web application by using only React and Vite, you need to do a lot of configurations and optimizations. 
But with Rasengan.js, you don't need to worry about that. It is already optimized for performance and provides a simple and easy-to-use API that will help you to create a high-quality web application.

Rasengan.js works on top of the Server Side Rendering (SSR) which is useful for SEO and performance optimization. It also provides a simple and easy-to-use [`Data Fetching`](https://rasengan.dev/docs/core/rendering/data-fetching) system that allows you to fetch data from an API (similar to `getServerSideProps` of Next.js).

### Prerequisites

Before you start, you need to have [Node.js](https://nodejs.org) installed on your machine. Prefer using the latest LTS version from version `v18.19` or later.

### Installation

To get started with Rasengan.js, you can follow the [installation guide](https://rasengan.dev/docs/getting-started/installation).

```bash filename="Terminal" copy
npx create-rasengan@latest my-app
```

This command will create a new Rasengan.js project in the `my-app` directory through simple steps.

## Documentation

The documentation of Rasengan.js is very clear and easy to understand. You can find the documentation [here](https://rasengan.dev/docs).

## Any question ?

Join our community and ask your questions in the [Github Discussions](https://github.com/rasengan-dev/rasenganjs/discussions), we will be happy to help you.

## Conclusion

Rasengan.js is very powerful, it provides you a new and simple approach to create a high-quality web application. It is built with modern tools like React, Vite, and React Router, which makes it a great choice for your web application.

Don't hesitate to give it a try and let us know your feedback. We are always open to suggestions and improvements. Contact us at [reactgx@gmail.com](mailto:reactgx@gmail.com)

**Comment by Coding-Industry-Dev** [comment]
it's great !! We are going to test it

**Reply by dilane3** [reply]
Nice 🙂

Drop your experience with the tool here, when possible.
